C/tasks/north/sending_process/sending.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * FogLAMP process class
+ *
+ * Copyright (c) 2018 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto
+ */
+
+#include <sending.h>
+
+//Define the type of the plugin managed by the Sending Process
+#define PLUGIN_TYPE "north"
+
+// Types of sources for the data blocks
+#define DATA_SOURCE_READINGS   "readings"
+#define DATA_SOURCE_STATISTICS "statistics"
+#define DATA_SOURCE_AUDIT      "audit"
+
+// Audit code to use
+#define AUDIT_CODE "STRMN"
+
+// Configuration retrieved from the Configuration Manager
+#define CONFIG_CATEGORY_NAME "SEND_PR"
+#define CONFIG_CATEGORY_DESCRIPTION "Configuration of the Sending Process"
+
+// Buffer max elements
+#define DATA_BUFFER_ELMS 10
+
+using namespace std;
+
+// Mutex for m_buffer access
+std::mutex	cReadMutex;
+
+// Destructor
+SendingProcess::~SendingProcess()
```

Comment:
I would suggest adding private copy constructor and operator= method to prevent accidental creation of copies that led to issue with the memory allocation.

Reply:
Done